server/protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java

```diff
@@ -71,9 +68,10 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
         try {
             HttpServletRequest requestWithSession = authMethods.stream()
-                    .filter(auth -> auth.checkAuthorizationHeader(getAuthHeaders(httpRequest)))
+                    .map(auth -> createSession(auth, httpRequest))
+                    .filter(Optional::isPresent)
                     .findFirst()
-                    .map(auth -> addSessionToRequest(httpRequest, createSession(auth, getAuthHeaders(httpRequest))))
+                    .map(mailboxSession -> addSessionToRequest(httpRequest, mailboxSession))
```

Comment:
if you put  `map`  before `findFirst` you would not get a Optional<MailboxSession> as addSessionToRequest argument